src/domManipulation.ts

```diff
@@ -150,51 +186,72 @@ export function loadWeatherForecast(location, fetchResults) {
           tempList[1].innerText = `${maxTempResult} °C`;
-          tempList[1].style.backgroundColor = temperatureColourConverter(maxTempResult);
+          tempList[1].style.backgroundColor = temperatureColourConverter(Number(maxTempResult));
         }
       }
     }
   }
 }
 
 // Controls what happens when the network call has been unsuccessful.
-export function failedForecast(location) {
+export function failedForecast(location: string) {
   console.error(`Unable to fetch the forecast for the ${location} location`);
-  document.getElementById("popup-text").innerText = "Unable to fetch forecast";
+  const popupTextElement = document.getElementById("popup-text");
+  if (popupTextElement !== null) {
+    popupTextElement.innerText = "Unable to fetch forecast";
+  }
+  
 }
 
 // Hides the loader after the network call has been completed.
-export function hideForecastLoader(location) {
+export function hideForecastLoader(location: string) {
   if (location === "custom") {
-    document.getElementById("popup-loading-container").style.visibility = "hiddem";
-    document.getElementById("popup-loading-container").style.display = "none";
-    document.getElementById("popup-forecast-container").style.visibility = "visible";
-    document.getElementById("popup-forecast-container").style.display = "flex";
+
+    const popupLoader = document.getElementById("popup-loading-container");
+    if (popupLoader !== null) {
+      popupLoader.style.visibility = "hidden";
+      popupLoader.style.display = "none";
+    }
+   
+    const popupForecast =  document.getElementById("popup-forecast-container");
+
+    if (popupForecast !== null) {
+      popupForecast.style.visibility = "visible";
+      popupForecast.style.display = "flex";
+    }
+   
   }
 }
-export function handleMainCityClick(numButton) {
+export function handleMainCityClick(numButton: number) {
   const forecastDisplay = document.getElementsByClassName("city-forecast");
   // Remove the active state from the previously selected button.
+
   mainCityButtons[selectedMainCity].classList.remove("active");
   // Update the index of the newly/currently selected button.
   selectedMainCity = numButton;
   // Set the newly selected button to active.
   mainCityButtons[selectedMainCity].classList.add("active");
 
   // Display forecast for selected city and hide forecasts for other main cities/locations.
-  for (let j = 0; j < forecastDisplay.length; j++) {
-    if (j === selectedMainCity) {
-      // Display corresponding forecast of selected main city.
-      forecastDisplay[j].style.visibility = "visible";
-      forecastDisplay[j].style.display = "flex";
-    } else {
-      // Hide all other cities' information.
-      forecastDisplay[j].style.visibility = "hidden";
-      forecastDisplay[j].style.display = "none";
+  if (forecastDisplay !== null) {
+    for (let j = 0; j < forecastDisplay.length; j++) {
+      if (j === selectedMainCity) {
```

Comment:
could use `for ( let item of ....` instead, but it doesn't really make a difference

Reply:
const of ;)

Reply:
I am using the index to keep track of which city I have selected to know which one of the respective city's forecast containers I should make visible.